data_safe_haven/commands/sre.py

```diff
@@ -55,6 +56,17 @@ def deploy(
         )
         sre_config = SREConfig.from_remote_by_name(context, name)
 
+        # Check whether current IP address is authorised to take administrator actions
+        ip_address = current_ip_address(as_cidr=True)
+        if ip_address not in sre_config.sre.admin_ip_addresses:
+            msg = " ".join(
+                [
+                    f"Current IP address {ip_address} is not authorised to deploy SRE '[green]{sre_config.description}[/]'.",
+                    "Please update the 'admin_ip_addresses' section of the config file.",
+                ]
+            )
+            raise DataSafeHavenConfigError(msg)
+
```

Comment:
As there are two instances of this, can we put it in a function?

Do we want to raise `typer.Exit` here?